internal/terraform/terraform.go

```diff
@@ -0,0 +1,131 @@
+package terraform
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/hc-install/product"
+	"github.com/hashicorp/hc-install/releases"
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+type runner struct {
+	installer  *releases.LatestVersion
+	runner     *tfexec.Terraform
+	workingDir string
+}
+
+// New creates a terraform runner for the provided working directory where
+// terraform files are. It returns a TerraformRunner struct which has the
+// runner to perform any commands terraform-exec package provides.
+func New(ctx context.Context, workingDir string) (*runner, error) {
+	installer := releases.LatestVersion{
+		Product:    product.Terraform,
+		InstallDir: "/tmp",
+	}
+
+	execPath, err := installer.Install(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error installing terraform: %w", err)
+	}
+
+	tf, err := tfexec.NewTerraform(workingDir, execPath)
+	if err != nil {
+		return nil, fmt.Errorf("error configuring terraform: %w", err)
+	}
+
+	return &runner{
+		installer:  &installer,
+		runner:     tf,
+		workingDir: workingDir,
+	}, err
+}
+
+// SetEnvVars sets environment variables to be used when invoking terraform
+func (r *runner) SetEnvVars(envVars map[string]string) error {
+	env := make(map[string]string)
+
+	for _, item := range os.Environ() {
+		data := strings.Split(item, "=")
+		env[data[0]] = data[1]
+	}
+
+	for key, value := range envVars {
+		env[key] = value
+	}
+
+	return r.runner.SetEnv(env)
+}
+
+// Uninstalls the terraform instance installed at runtime
+func (r *runner) Uninstall(ctx context.Context) error {
+	err := r.installer.Remove(ctx)
+	if err != nil {
+		return fmt.Errorf("error uninstalling terraform: %w", err)
+	}
+
+	return nil
+}
+
+// Init performs a terraform init using the provided TerraformRunner receiver
+func (r *runner) Init(ctx context.Context) error {
+	err := r.runner.Init(ctx)
+	if err != nil {
+		return fmt.Errorf("error running terraform init: %w", err)
+	}
+
+	return nil
+}
+
+// Plan performs a terraform plan using the provided TerraformRunner receiver.
+func (r *runner) Plan(ctx context.Context, args ...tfexec.PlanOption) error {
+	args = append(args, tfexec.Out(fmt.Sprintf("%s/plan.out", r.workingDir)))
```

Comment:
I do like this idea, so its easy for a developer to quickly copy/paste and run from their environment.

Since we calling out to terraform-exec to handle running terraform, we will need to populate the terraform command to display. As right now, they do not provide back the os/exec command they used, where we could easily call the `string()` function of the command to print the entire command.

Reply:
I opened _\SDCICD-1043_ where we can track this suggestion and implement it in a separate PR.